lib/facility_factory.rb

```diff
@@ -0,0 +1,40 @@
+class FacilityFactory
```

Comment:
This class looks great! One way you could refactor just a little would be to keep only the formatting of the data in the individual state methods and then have one `create_facilities` method responsible for the `Facility.new....` part of the transaction. Overall I like this a lot though and that you've got each state separated out to its own method rather than a giant if statement.